src/meta-srv/src/procedure/region_follower/add_region_follower.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_meta::instruction::CacheIdent;
+use common_procedure::error::ToJsonSnafu;
+use common_procedure::{
+    Context as ProcedureContext, Error as ProcedureError, LockKey, Procedure,
+    Result as ProcedureResult, Status,
+};
+use common_telemetry::info;
+use snafu::ResultExt;
+use store_api::storage::RegionId;
+
+use super::create::CreateFollower;
+use super::{AlterRegionFollowerData, AlterRegionFollowerState, Context};
+use crate::error::{self, Result};
+use crate::metrics;
+pub struct AddRegionFollowerProcedure {
+    pub data: AlterRegionFollowerData,
+    pub context: Context,
+}
+
+impl AddRegionFollowerProcedure {
+    pub const TYPE_NAME: &'static str = "metasrv-procedure::AddRegionFollower";
+
+    pub fn new(
+        catalog: String,
+        schema: String,
+        region_id: RegionId,
+        peer_id: u64,
+        context: Context,
+    ) -> Self {
+        Self {
+            data: AlterRegionFollowerData {
+                catalog,
+                schema,
+                region_id,
+                peer_id,
+                peer: None,
+                datanode_table_value: None,
+                table_route: None,
+                state: AlterRegionFollowerState::Prepare,
+            },
+            context,
+        }
+    }
+
+    pub fn from_json(json: &str, context: Context) -> ProcedureResult<Self> {
+        let data: AlterRegionFollowerData = serde_json::from_str(json).unwrap();
+        Ok(Self { data, context })
+    }
+
+    pub async fn on_prepare(&mut self) -> Result<Status> {
+        // loads the datanode peer and check peer is alive
+        let datanode_peer = self.data.load_datanode_peer(&self.context).await?;
+
+        // loads the datanode table value
+        let _ = self.data.load_datanode_table_value(&self.context).await?;
```

Comment:
The side effect of the function is a little ambigous to me. I prefer to explicitly use
```rust
self.context.data.some_member = self.data.load_datanode_table_value(&self.context).await?;
```